### ISSUE TITLE
Payload type unknown when defining optional property?

### ISSUE THREAD

**Issue by mcchrish** [reduxjs/redux-starter-kit#214]
This is for version 0.8.0.

```js
const action = createAction<{ input?: string }>('ACTION')
type ActionType = ReturnType<typeof action> // ActionType.payload === unknown


const action = createAction<{ input?: string, input2: string }>('ACTION')
type ActionType = ReturnType<typeof action> // ActionType.payload === { input?: string | undefined; input2: string }
```

Somehow, if there is only one property and it's optional, it converts it type to `unknown`

**Comment by markerikson** [comment]
Oh dear.

@phryneas , @jessidhia : any ideas?

**Comment by phryneas** [comment]
Phew. Yup.

The feature detection for TS<3.5 was leaking both cases (TS<3.5 and TS>=3.5), exposing the pre-3.5 behaviour. 
Back then, we could not distinguish between the empty object default case and `unknown`, so it would fall back to `unknown` behaviour if the object looked like an empty object (which is the case here).

We can fix this for TS>=3.5, but in old versions this will be the behaviour if something possibly looks like an empty object.

( fixed TS version feature detection would look like this:
```typescript
export type AtLeastTS35<True, False> = [True, False][IsUnknown<ReturnType<<T>() => T>, 0, 1>]
```
)

Question is: is it worth it supporting old TS versions, which should be our minimum requirement? I guess we should still support 3.4, which is from march 2019, even if it hurts a little :/

**Comment by markerikson** [comment]
Yeah, 3.4 seems reasonable to support, even if it's a bit of a pain.

**Comment by phryneas** [comment]
I might be even able to fix it for pre-3.5... PR coming in.

**Comment by markerikson** [comment]
Fix published as 0.8.1.  Hopefully that should resolve this - let us know if there's still an issue!
